### PR TITLE
create helper window and add keyboard shortcuts - fix #7

### LINKED ISSUE #7: add helper menu to show short cuts and point to context menu functionality
As discussed with @radibit.
- [ ] add menu entry for `shortcuts` or anything
- [ ] open new window showing all available "hidden" functionalities ( this can be done the same way, as the about page )

### PR DESCRIPTION
**Pull request by radibit**
This PR provides solution for the following two items:
- [x] add menu entry for `shortcuts` or anything
- [x] open new window showing all available "hidden" functionalities ( this can be done the same way, as the about page )

In addition I extracted some repeated visual elements into components, following the proper namespace.

I'm open for suggestions for the actual content for the help window :)

### CODE REVIEW

File: app/main/menu.js [hunk 1]
```diff
@@ -72,6 +72,11 @@ module.exports = {
         role    : 'help',
         submenu : [
           {
+            label       : 'Keyboard Shortcuts',
+            accelerator : 'CmdOrCtrl+/',
+            click() { options.openHelpWindow(); }
+          },
+          {
             label : 'Learn More',
```
Comment (stefanjudis) [inline app/main/menu.js]: when we're already here - can we drop this `learn more`?

And maybe link to github issues? -> `Report a problem`

File: app/electron.js [hunk 2]
```diff
@@ -20,10 +21,12 @@ if ( process.env.NODE_ENV === 'development' ) {
   config          = require( '../config' ).config;
   config.url      = `http://localhost:${ config.port }`;
   config.aboutUrl = `http://localhost:${ config.port }#!/about`;
+  config.helpUrl  = `http://localhost:${ config.port }#!/help`;
 } else {
   config.devtron  = false;
   config.url      = `file://${ __dirname }/dist/index.html`;
   config.aboutUrl = `file://${ __dirname }/dist/index.html#!/about`;
+  config.helpUrl  = `file://${ __dirname }/dist/index.html#!/help`;
```
Comment (stefanjudis) [inline app/electron.js]: now that it's three rows maybe something like `baseUrl` makes sense?

To not repeat the paths over and over again?

File: app/src/App.vue [hunk 3]
```diff
@@ -8,6 +8,8 @@
   @import './styles/objects/paragraphs';
   @import './styles/objects/lists';
   @import './styles/objects/small';
+  @import './styles/components/drag-handle';
+  @import './styles/components/logo';
```
Comment (stefanjudis) [inline app/src/App.vue:12]: sweet! 🍠

**Comment by stefanjudis** [comment]
@radibit Man, nice PR.

Things I'd like to document as well.
- `ESC` closing a running command
-  `->` executes a given command when in project panel
- `<-` terminates a running command 
- `right click` on projects brings more options

Not sure if grouping by panel makes sense... But it was like this in my head somehow. :)

Overall - really really nice. 👍

File: app/src/components/HelpView.vue [hunk 4]
```diff
@@ -0,0 +1,127 @@
...
+            <feGaussianBlur stdDeviation="1.5" in="SourceAlpha" result="shadowBlurInner1" />
+            <feOffset dy="1" in="shadowBlurInner1" result="shadowOffsetInner1" />
+            <feComposite in="shadowOffsetInner1" in2="SourceAlpha" operator="arithmetic" k2="-1" k3="1" result="shadowInnerInner1"
+            />
+            <feColorMatrix values="0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0.5 0" in="shadowInnerInner1" />
+          </filter>
+        </defs>
+        <g fill="none" fill-rule="evenodd">
+          <g transform="translate(4 3)">
+            <use fill="#000" filter="url(#a)" xlink:href="#b" />
```
Comment (stefanjudis) [inline app/src/components/HelpView.vue]: I see you put a lot of effort in the svg things. :)

These are the first things that use gradients and shadows... For me they're no matching 100% the interface ( 90% ) and they add more complexity to this view...

Maybe we can live without them?

File: app/src/components/HelpView.vue [hunk 5]
```diff
@@ -0,0 +1,127 @@
...
+        </g>
+      </svg>
+    </div>
+    <div class="u-marginBottom">
+      <h1 class="o-headline-2">Keyboard Shortcuts</h1>
+    </div>
+
+    <dl class="c-help--definitionList">
+      <dt>
+        <span class="c-help--key">&#8592;</span>
```
Comment (stefanjudis) [inline app/src/components/HelpView.vue]: a definition list without words...

Maybe we can use `aria-label` here?

File: app/electron.js [hunk 6]
```diff
@@ -59,6 +62,23 @@ function openAboutWindow() {
   }
 }
 
+function openHelpWindow() {
```
Comment (stefanjudis) [inline app/electron.js]: also this is pretty much the same as about... 

I wonder if we can't unify them with some smart config?

Maybe something like

Pseudo coded:

```
const staticWindows = {
  help : {
    config : {}, // the stuff for browserWindow
    hash : '#help',
    initializedWindow : null
  },
  // ...
}

function openStaticWindow( type ) {
  if ( ! staticWindows[ type ].initializedWindow ) {
     staticWindows[ type ].initializedWindow = new BrowserWindow( staticWindows[ type ].config );

     staticWindows[ type ].initializedWindow.loadURL( staticWindows[ type ].url );

     staticWindows[ type ].initializedWindow.on( 'closed', () => staticWindows[ type ].initializedWindow = null );
   } else {
     staticWindows[ type ].initializedWindow.focus();
   }
}
```

**Comment by radibit** [comment]
@stefanjudis Please have a look on the changes we discussed and if they are acceptable as part of this PR :)

**Comment by stefanjudis** [comment]
@radibit your build failed. :)

forrest-help

Just fiddled around a bit... What do you think, it looks a bit cleaner and saves space.

Uups... has to be `navigate through commands`.

And left arrow on project code be `Go home / Terminate running command`
And maybe `keyboard shortcuts` should be only `shortcuts` as a mouse click is in there.

Bonus: Can you please add the "Report a problem" menu entry? And a rebase with one sweet commit would be cool. :)

**Comment by radibit** [comment]
@stefanjudis after applying all the changes and fixing reported issues, I tried to squash all the commits, but unfortunately I can't see all the selected commits as one, though the operation was successful :/ I'd be happy if you can share what actually is preventing me of merging them. 

Also please verify and confirm the latest changes, hope we reached the desired state :)

File: app/electron.js [hunk 7]
```diff
@@ -21,8 +21,8 @@ const staticWindows = {
   },
   help : {
     config : {
-      height          : 625,
-      width           : 625,
+      height          : 400,
+      width           : 550,
```
Comment (stefanjudis) [inline app/electron.js:25]: @radibit This was a missunderstanding. Was too lazy in sketch to adjust that. :)

Comment (radibit) [inline app/electron.js:25]: yeah, no problem I was thinking the same, but was not sure if we want to add more info now.

**Comment by stefanjudis** [comment]
to keep commit history clean i opened https://github.com/stefanjudis/forrest/pull/40 and squashed everything there.

**Pull request closed without merging**